core/src/main/scala/scala/pickling/generator/sourcegen.scala

```diff
@@ -400,6 +396,11 @@ private[pickling]  trait SourceGenerator extends Macro with tags.FastTypeTagMacr
         implicit object $picklerName extends _root_.scala.pickling.Pickler[$tpe] with _root_.scala.pickling.Generated {
           def pickle(picklee: $tpe, builder: _root_.scala.pickling.PBuilder): _root_.scala.Unit = ${genPicklerLogic[T](picklerAst)}
           val tag: _root_.scala.pickling.FastTypeTag[$tpe] = $createTagTree
+
+          if(_root_.scala.pickling.internal.currentRuntime.picklers.lookupPickler(tag.key).isEmpty) {
```

Comment:
Well, no, actually I have an idea to avoid the instantiation. I'll try it out in my next commit.